### PR TITLE
batch refactoring

### PR DESCRIPTION
**Pull request by jasonpilz**
What's this PR do?
- Mainly fixes syntax to use es6 features consistently, including es6 arrow functions, string templates, insert missing semi-colons, change remaining `var`s to `let`s.
- Remove duplicate README and un-needed .coffee file.
- Checked using jshint.
- Added `.jshintrc`.

Where should the reviewer start?
- All `.js` files were refactored, can start on anyone.
  How should this be manually tested?
- Currently no tests implemented, however syntax changes were checked with jshint.

Any background context you want to provide?
- General refactoring to maintain consistent style choices.

Screenshots (if appropriate)
- None available

What gif best describes this PR or how it makes you feel?
- Feels good to have less-smelly code

@rrgayhart
@stevekinney

### CODE REVIEW

**Comment by jasonpilz** [comment]
closes #13

File: app/assets/javascripts/delete.js.es6 [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
         $.ajax({
             type: 'DELETE',
             url: '/api/v1/ideas/' + $idea.attr('idea-id'),
-            success: () => { $idea.parent().remove() },
-            error: (xhr) => { alert(xhr.responseTest) }
-        })
-    })
-}
+            success: () => { $idea.parent().remove(); },
+            error: (xhr) => { alert(xhr.responseTest); }
```
Comment (stevekinney) [inline app/assets/javascripts/delete.js.es6:14]: I still feel like either the callback or the promise-style pattern is a better fit here.

Comment (rrgayhart) [inline app/assets/javascripts/delete.js.es6:14]: @stevekinney - can you give a code example of what you mean here?

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (rrgayhart) [inline .jshintrc:1]: Hooray for jshint wrangling!

**Comment by rrgayhart** [comment]
:+1: Looks great! Good choice for refactoring here!

**Pull request merged**